### PR TITLE
KubernetesPodOperator Guide

### PR DESCRIPTION
**Pull request by VBhojawala**
Added KubernetesPodOperator guide in KubernetesPodOperator.rst

related: #8970

### CODE REVIEW

**Comment by john-jac** [comment]
Hi @VBhojawala !  Just reviewing your doc and it looks good.  FYI there's one focused on Amazon MWAA and EKS at https://docs.aws.amazon.com/mwaa/latest/userguide/mwaa-eks-example.html that might be worth a reference as it, too, simplifies the authorization process.

**Comment by VBhojawala** [comment]
Hi @dimberman,
Kindly  review the PR.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 1]
```diff
@@ -15,114 +15,1015 @@
...
 
 
 .. _howto/operator:KubernetesPodOperator:
 
-KubernetesPodOperator
-=====================
+Kubernetes Pod Operator "Hello World!"
+======================================
+
+- Creating Task using KubernetesPodOperator which prints "Hello World [HOSTNAME]".  Add ``in_cluster=False`` to
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: Hi Vivek, could you put examples of using pod_template file and pod_template_spec on top? Both of those could make this operator a lot easier to use.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 2]
```diff
@@ -15,114 +15,1015 @@
...
+  KubernetesPodOperator constructor when running examples with minikube.
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: I'd recommend we push people towards KinD instead of minikube.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 3]
```diff
@@ -15,114 +15,1015 @@
...
+      task1 = KubernetesPodOperator(task_id='k8s_pod_operator_task',
+                                    name='airflow_pod_operator',
+                                    namespace='default',
+                                    image='alpine',
+                                    cmds=["sh", "-c",
+                                          'echo "Hello World from pod [$HOSTNAME]"'],
+                                    startup_timeout_seconds=60,
+                                    )
+
+- Executing / Debugging example and checking the logs
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: Is there something you were going to write here? I just see logs...

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 4]
```diff
@@ -15,114 +15,1015 @@
...
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-aed97ecd64854367ad7d0ff39f37859c Succeeded
+  {taskinstance.py:1136} INFO - Marking task as SUCCESS. dag_id=example_k8s_operator, task_id=k8s_pod_operator_task,
+  execution_date=20201203T000000, start_date=20201204T140331, end_date=20201204T140345
+  ................................................................................................................
+
+- Getting kubernetes pods using labels ``dag_id`` and ``task_id`` automatically assigned by Airflow and Describing it.
+
+.. code-block:: bash
+
+  $ kubectl get pods -l dag_id=example_k8s_operator,task_id=k8s_pod_operator_task
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:262]: good catch on this one!

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 5]
```diff
@@ -15,114 +15,1015 @@
...
-and type safety. While we have removed almost all Kubernetes convenience classes, we have kept the
-:class:`~airflow.kubernetes.secret.Secret` class to simplify the process of generating secret volumes/env variables.
-
-.. exampleinclude:: /../../airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.py
-    :language: python
-    :start-after: [START howto_operator_k8s_cluster_resources]
-    :end-before: [END howto_operator_k8s_cluster_resources]
-
-Difference between ``KubernetesPodOperator`` and Kubernetes object spec
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:77]: Why are you getting rid of this?

**Review by dimberman — CHANGES_REQUESTED** [review]
Hi Vivek, 

I'm seeing a lot of great examples here but I'm confused to why you are deleting all of the text describing things like xcom/private images/ pod_template etc.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 1 above)
Comment (VBhojawala) [reply docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: Ok , I will put that example of top.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 2 above)
Comment (VBhojawala) [reply docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: ok.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 3 above)
Comment (VBhojawala) [reply docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: I meant after executing the dag with configuration, i will update the line.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 5 above)
Comment (VBhojawala) [reply docs/apache-airflow-providers-cncf-kubernetes/operators.rst:77]: I will added text description from old doc.

**Comment by VBhojawala** [comment]
Hi @dimberman ,
Thank you for reviewing PR>
I have added text from previous document and made changes suggested in review.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 6]
```diff
@@ -115,14 +905,169 @@ alongside the Pod. The Pod must write the XCom value into this location at the `
...
-    :start-after: [START howto_operator_k8s_write_xcom]
-    :end-before: [END howto_operator_k8s_write_xcom]
 
-Reference
-^^^^^^^^^
-For further information, look at:
+- Example Dag : KubernetesPodOperator task writes contents to be returned to ``/airflow/xcom/return.json`` and reading
+  values returned using ``xcom_pull(key, task_ids)``.
+
+.. code-block:: python
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: Can you please move these code segments into example_dag files or some other external file and pull in using tags? That way we can re-use these code snippets later and have python linting when editing them.

**Review by dimberman — CHANGES_REQUESTED** [review]

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 6 above)
Comment (VBhojawala) [reply docs/apache-airflow-providers-cncf-kubernetes/operators.rst]: I have extracted example and tagged in doc.

**Comment by VBhojawala** [comment]
Hi @dimberman,
Kindly review the PR.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 7]
```diff
@@ -15,28 +15,187 @@
...
 
-KubernetesPodOperator
-=====================
 
-The :class:`~airflow.providers.cncf.kubernetes.operators.kubernetes_pod.KubernetesPodOperator` allows
-you to create and run Pods on a Kubernetes cluster.
+KubernetesPodOperator with YAML file / JSON Spec
+================================================
+
+- Creating a Pod Template file
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]:
```suggestion
Below is an example of a basic pod_template_file for a KubernetesPodOperator pod. Please note that the first container is named base, as this is required for Airflow to later run the pod
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 8]
```diff
@@ -15,28 +15,187 @@
...
+            cpu: 1
+            memory: "100Mi"
+        command: ['sh', '-c', 'echo "myapp-pod created from YAML pod template."']
+    volumes:
+      - name: myapp-volume
+        persistentVolumeClaim:
+          claimName: myapp-pvc-rw
+
+
+- Creating pod from a template using KubernetesPodOperator
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]:
```suggestion
Once you have created a pod_template_file, you can use this file as a basis for your KPO pod, while using the other supplied arguments as "overrides". In this example,, we are able to take this template and modify the namespace and name of the pod.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 9]
```diff
@@ -15,28 +15,187 @@
...
 
-.. note::
-  If you use `Google Kubernetes Engine <https://cloud.google.com/kubernetes-engine/>`__, consider
-  using the
-  :ref:`GKEStartPodOperator <howto/operator:GKEStartPodOperator>` operator as it
-  simplifies the Kubernetes authorization process.
 
-.. note::
-  The :doc:`Kubernetes executor <apache-airflow:executor/kubernetes>` is **not** required to use this operator.
+- Examining pod configuration
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]:
```suggestion
If you need to examine a pod that is not running correctly, run the command ``kubectl describe pod <my pod name> to get all relevant details about the pod from the k8s cluster.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 10]
```diff
@@ -15,28 +15,187 @@
...
-.. contents::
-  :depth: 1
-  :local:
+  {taskinstance.py:1136} INFO - Marking task as SUCCESS. dag_id=example_k8s_yaml_config, task_id=
+  k8s_pod_yaml_config_task, execution_date=20201205T000000, start_date=20201206T130803, end_date=20201206T130818
+  {taskinstance.py:1195} INFO - 0 downstream tasks scheduled from follow-on schedule check
+  {dagrun.py:447} INFO - Marking run <DagRun example_k8s_yaml_config @ 2020-12-05 00:00:00+00:00:
+  backfill__2020-12-05T00:00:00+00:00, externally triggered: False> successful
+  {backfill_job.py:377} INFO - [backfill progress] | finished run 1 of 1 | tasks waiting: 0 | succeeded: 1 |
+  running: 0 | failed: 0 | skipped: 0 | deadlocked: 0 | not ready: 0
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:111]: I'm not sure what the point of this section is? Are you showing what the logs look like? If so can you give instructions on how to find the logs (e.g. via the UI or via a k8s command)

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 11]
```diff
@@ -53,25 +212,120 @@ dependencies that are not available through the public PyPI repository. It also
...
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-aed97ecd64854367ad7d0ff39f37859c Succeeded
+
+  {pod_launcher.py:136} INFO - Hello World from pod [airflow-pod-operator-aed97ecd64854367ad7d0ff39f37859c]
+
+  {pod_launcher.py:176} INFO - Event: airflow-pod-operator-aed97ecd64854367ad7d0ff39f37859c had an event of
+  type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-aed97ecd64854367ad7d0ff39f37859c Succeeded
+  {taskinstance.py:1136} INFO - Marking task as SUCCESS. dag_id=example_k8s_operator, task_id=k8s_pod_operator_task,
+  execution_date=20201203T000000, start_date=20201204T140331, end_date=20201204T140345
+  ................................................................................................................
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:254]: Why did you add more log lines here?

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 12]
```diff
@@ -53,25 +212,120 @@ dependencies that are not available through the public PyPI repository. It also
...
+
+- Getting kubernetes pods using labels ``dag_id`` and ``task_id`` automatically assigned by Airflow and Describing it.
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]:
```suggestion
Another option for getting the pod launched by Airflow is to take advantage of the labels that Airflow injects into every KubernetesPodOperator pod. Using the line below, you can quickly get the name of the pod launched by Airflow.

``kubectl get pods -l dag_id=<your dag id>,task_id=<your task id>``
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst (hunk 11 above)
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:254]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 13]
```diff
@@ -53,25 +212,120 @@ dependencies that are not available through the public PyPI repository. It also
 YAML file using the ``pod_template_file`` parameter.
 Ultimately, it allows Airflow to act a job orchestrator - no matter the language those jobs are written in.
 
-How to use cluster ConfigMaps, Secrets, and Volumes with Pod?
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
-To add ConfigMaps, Volumes, and other Kubernetes native objects, we recommend that you import the Kubernetes model API
-like this:
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:60]: Why did you remove this section?

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 14]
```diff
@@ -53,25 +212,120 @@ dependencies that are not available through the public PyPI repository. It also
...
+                     node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
+    Events:
+      Type    Reason     Age   From               Message
+      ----    ------     ----  ----               -------
+      Normal  Scheduled  15m   default-scheduler  Successfully assigned default/airflow-pod-operator
+                                                  -aed97ecd64854367ad7d0ff39f37859c to minikube
+      Normal  Pulling    15m   kubelet            Pulling image "alpine"
+      Normal  Pulled     15m   kubelet            Successfully pulled image "alpine" in 4.214686688s
+      Normal  Created    15m   kubelet            Created container base
+      Normal  Started    15m   kubelet            Started container base
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:329]:
```suggestion
```

We don't need to show this twice

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 15]
```diff
@@ -53,25 +212,120 @@ dependencies that are not available through the public PyPI repository. It also
...
 
-.. exampleinclude:: /../../airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.py
-    :language: python
-    :start-after: [START howto_operator_k8s_cluster_resources]
-    :end-before: [END howto_operator_k8s_cluster_resources]
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:74]: Why did you remove this?

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 16]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+                                    namespace='default',
+                                    env_vars=env_vars,
+                                    image='alpine',
+                                    cmds=["sh", "-c",
+                                          'echo "Reading environment variables TIME_OUT : $TIME_OUT   ENV_TYPE :'
+                                          ' $ENV_TYPE"'],
+                                    startup_timeout_seconds=60,
+                                    )
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:372]: Can you put this is an example python file?

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 17]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+  {pod_launcher.py:176} INFO - Event: airflow-pod-operator-env-3824c08cb2f04af7928103a027189668 had an
+  event of type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-env-3824c08cb2f04af7928103a027189668 Succeeded
+  {pod_launcher.py:176} INFO - Event: airflow-pod-operator-env-3824c08cb2f04af7928103a027189668 had an event
+  of type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-env-3824c08cb2f04af7928103a027189668 Succeeded
+  ...................................................................................................................
+
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:401]:
```suggestion
```

We don't need to show logs for all of these.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 18]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+Accessing ConfigMap
+===========================
+
+
+- YAML file for creating ConfigMap in Kubernetes
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:406]:
```suggestion
If you would like to use a configMap in your kubernetes deployment, first add a configMap to your namespace before launching the pod. Below is an example of a basic configMap.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 19]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+    ----
+    15
+    ENV_TYPE:
+    ----
+    test
+    Events:  <none>
+
+
+
+- Accessing variables from ConfigMap inside the Pod
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:445]:
```suggestion
Once you have created this configmap, you can access it using k8s python client objects. Below is an example of adding a configMap to a KubernetesPodOperator deployment.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 20]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+      task1 = KubernetesPodOperator(task_id='k8s_pod_operator_config_map_task',
+                                    name='airflow_pod_operator_config_map',
+                                    namespace='default',
+                                    image='alpine',
+                                    env_from=config_map,
+                                    cmds=["sh", "-c",
+                                          'echo "Reading environment variables TIME_OUT : $TIME_OUT   ENV_TYPE :'
+                                          ' $ENV_TYPE"'],
+                                    startup_timeout_seconds=60,
+                                    )
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:469]: Please move this into its own python file and then reference it, unless @kaxil or @potiuk thinks it's ok to keep small snippets in documentation.

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 21]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-config-map-d472d9495b0741bc84e34c11d09c66fe Succeeded
+
+  {pod_launcher.py:136} INFO - Reading environment variables TIME_OUT : 15   ENV_TYPE : test
+
+  {pod_launcher.py:176} INFO - Event: airflow-pod-operator-config-map-d472d9495b0741bc84e34c11d09c66fe had an
+  event of type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-config-map-d472d9495b0741bc84e34c11d09c66fe Succeeded
+
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:496]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 22]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+Accessing Secrets
+=================
+
+
+- Creating Secrets
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:501]:
```suggestion
To test adding secrets to a KubernetesPodOperator pod, we first create a basic secret using the steps below.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 23]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+
+    Type:  Opaque
+
+    Data
+    ====
+    DB_PWD:   13 bytes
+    DB_USER:  4 bytes
+
+
+- Accessing secret inside pod as environment variable
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:525]:
```suggestion
Now that this secret has been added to our cluster, we can access it using the following code
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 24]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+  AIRFLOW_CTX_DAG_RUN_ID=backfill__2020-12-04T00:00:00+00:00
+
+  {pod_launcher.py:136} INFO - Reading environment variables DB_USER: : root:   DB_PWD : ent3r$ce@d00r
+
+  {pod_launcher.py:176} INFO - Event: airflow-pod-operator-secret-3f7d6b3e5dcf4673aa1f584e26f1d012 had an
+  event of type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-operator-secret-3f7d6b3e5dcf4673aa1f584e26f1d012 Succeeded
+
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:569]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 25]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+Mounting Persistent  Volume to Pod
+==================================
+
+- Creating PersistentVolume
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:573]:
```suggestion
To add a PersistentVolume to your KubernetesPodOperator deployment, start by adding a PersistentVolume and a PersistentVolumeClaim as seen below
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 26]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+    hostPath:
+      path: /tmp/myapp
+
+
+.. code-block:: bash
+
+  $ kubectl apply -f myapp_pv.yaml
+
+
+- Creating PersistentVolumeClaim
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:597]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 27]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+    - ReadWriteMany
+    storageClassName: ""
+
+
+.. code-block:: bash
+
+  $ kubectl apply -f myapp_pvc.yaml
+
+
+- Writing and Reading file from Persistent Volume using KubernetesPodOperator
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:619]:
```suggestion
Now that we have created the PersistentVolume and PersistentVolumeClaim, we can both read and write to the volume using the code below
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 28]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+  {pod_launcher.py:136} INFO - Reading date from date.txt : Sat Dec 5 13:58:35 UTC 2020
+
+  {pod_launcher.py:176} INFO - Event: airflow-pod-volume-write-23495d6738994e1d96765dfef49f345c had an event
+  of type Succeeded
+  {pod_launcher.py:289} INFO - Event with job id airflow-pod-volume-write-23495d6738994e1d96765dfef49f345c Succeeded
+  {backfill_job.py:377} INFO - [backfill progress] | finished run 1 of 1 | tasks waiting: 0 | succeeded: 2 |
+  running: 0 | failed: 0 | skipped: 0 | deadlocked: 0 | not ready: 0
+  {backfill_job.py:830} INFO - Backfill done. Exiting.
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:704]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 29]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+Mounting Secrets as Volume
+==========================
+
+- Example Dag demonstrating use of ``Secret`` class which internally configures ``Volume`` and ``VolumeMount`` for
+  given secret.
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:709]:
```suggestion
This example shows how we can use an Airflow ``Secret`` object to inject a secret into a pod as a VolumeMount
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 30]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+
+  {taskinstance.py:1195} INFO - 0 downstream tasks scheduled from follow-on schedule check
+  {dagrun.py:444} INFO - Marking run <DagRun example_k8s_secret_volume
+  @ 2020-12-06 00:00:00+00:00: backfill__2020-12-06T00:00:00+00:00, externally triggered: False> successful
+  {backfill_job.py:377} INFO - [backfill progress] | finished run 1 of 1 | tasks waiting: 0 | succeeded: 1 |
+  running: 0 | failed: 0 | skipped: 0 | deadlocked: 0 | not ready: 0
+  {backfill_job.py:830} INFO - Backfill done. Exiting.
+
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:775]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 31]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+  {pod_launcher.py:136} INFO - Hello World from pod [airflow-pod-limit-resource-ac4c107dd10549c89a2015f976e729d6]
+
+  {taskinstance.py:1136} INFO - Marking task as SUCCESS. dag_id=example_k8s_limit_resource,task_id=
+  k8s_pod_limit_resource_task, execution_date=20201205T000000, start_date=20201206T035456, end_date=20201206T035517
+  {backfill_job.py:377} INFO - [backfill progress] | finished run 1 of 1 | tasks waiting: 0 | succeeded: 1 |
+  running: 0 | failed: 0 | skipped: 0 | deadlocked: 0 | not ready: 0
+  {backfill_job.py:830} INFO - Backfill done. Exiting.
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:841]:
```suggestion
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 32]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+
+Defining Pod resource Requirements and Limits
+=============================================
+
+
+- Defining ResourceRequirements for Pod
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst]:
```suggestion
This example defines ResourceRequirements for Pod, which allow users to modify CPU, memory, and GPU requests and limits.
```

File: docs/apache-airflow-providers-cncf-kubernetes/operators.rst [hunk 33]
```diff
@@ -82,29 +336,565 @@ equivalent YAML/JSON object spec for the Pod you would like to run.
...
+          memory:     100Mi
+
+        Environment:  <none>
+        Mounts:
+          /var/run/secrets/kubernetes.io/serviceaccount from default-token-ltgdm (ro)
+    ..................................................................................
+
+
+
+
```
Comment (dimberman) [inline docs/apache-airflow-providers-cncf-kubernetes/operators.rst:897]:
```suggestion
```

**Review by dimberman — CHANGES_REQUESTED** [review]
Hi @VBhojawala I added some more descriptions and removed logs for most of the cases except special ones (e.g. showing how XCom works), please add these fixes and then I'll review this one more time. Also interested what @kaxil and @potiuk think about whether it's ok to keep code snippets in docs (I'm not super opinionated but I recall people telling me not to do that in the past)

**Comment by eladkal** [comment]
@VBhojawala can you take a look at the failing doc build?
